snuba/query/processors/hexint_column_processor.py

```diff
@@ -0,0 +1,123 @@
+from typing import Set
+
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.clickhouse.query import Query
+from snuba.query.conditions import ConditionFunctions, FUNCTION_TO_OPERATOR
+from snuba.query.exceptions import ValidationException
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.matchers import Or, Param, String
+from snuba.query.matchers import Any as AnyMatch
+from snuba.query.matchers import Column as ColumnMatch
+from snuba.query.matchers import FunctionCall as FunctionCallMatch
+from snuba.query.matchers import Literal as LiteralMatch
+from snuba.request.request_settings import RequestSettings
+
+
+class HexIntValidationError(ValidationException):
+    pass
+
+
+class HexIntColumnProcessor(QueryProcessor):
```

Comment:
This is largely a copy/paste from uuid_column_processor.
It seems the only differences are:
- the pattern for `col_func`
- the implementation of the translation function `translate_uuid_literal` there and `translate_hexint_literal` (both cases have the same interface)

I suggest we avoid this duplication.
A couple of ways:
- composition. Have a typed column processor that takes in the pattern and the translation function (or better, a class that does matching and translation on the same line as typed context promoter https://github.com/getsentry/snuba/blob/master/snuba/query/processors/typed_context_promoter.py#L29. There we would build the processor with objects of a class that would take care of a specific type)
- inheritance. Have a parent processor that runs the show and the extending classes that provide the pattern to match and the translation function.

I prefer the first.

Reply:
Good point, updated